### ISSUE TITLE
Autoinsurance can add an extreme amount of ammunition for free under specific circumstances

### ISSUE THREAD

**Issue by Caenen** [Freelancer-Sirius-Revival/FLSR-Hook#41]
Uncertain what the specific setup for this bug is at this time. On a freighter using both Drone Mines and Javelin Missiles, returning with both ammo stacks below insured amounts my Drone mine ammo was refilled to my cargo capacity and the remaining ammo return loops cancelled due to full cargo hold. My insurance return was 0 (due to the cost of the Drone mines being subtracted), but not less than 0, and selling the excess Drone mine ammo gave much more money than the insurance deposit was in the first place.

Below is the inventory before and after docking with this bug occuring, for reference.
![grafik](https://github.com/Freelancer-Sirius-Revival/FLSR-Hook/assets/79576409/9dbae48e-f95e-4fc2-9e2b-182ac9b47b4f)
![grafik](https://github.com/Freelancer-Sirius-Revival/FLSR-Hook/assets/79576409/3883dec0-3082-422c-937e-ab260bbcd1ca)

**Comment by erikroe** [comment]
@Caenen can you verify if this is still the case? I think we made a fix for that.

**Comment by Caenen** [comment]
We did fix that and I verified the fix on the test server. It's https://github.com/Freelancer-Sirius-Revival/FLSR-Hook/commit/8ad44d8cb47e806bb2f55fdc1364d03771b82721 but we didn't close the issue when you did the commit.

The issue was that, when landing with a larger stack of an item than it was insured for, it would find a negative value to restore and didn't properly handle that, "restoring" instead as much as cargo space was available for from that item, using up the insurance fund until it was 0, with any additional restored ammo becomign effectively free.

The fix was pushed to the live server at the time and has been in effect since.
